src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java

```diff
@@ -70,7 +74,35 @@ public CassandraClient connect(String keyspace, Handler<AsyncResult<Void>> conne
       }
     }
 
-    Cluster build = builder.withPort(options.port()).build();
+    Cluster build = builder.withNettyOptions(
+      new NettyOptions() {
+        @Override
+        public EventLoopGroup eventLoopGroup(ThreadFactory threadFactory) {
+          return vertx.getEventLoopGroup();
+        }
+
+        @Override
+        public void onClusterClose(EventLoopGroup eventLoopGroup) {
+          // it is important to not do anything here
+          // because the default behaviour is to shutdown the Vert.x event loop group
+        }
+
+        @Override
+        public Timer timer(ThreadFactory threadFactory) {
+          // we don't wan't to use the Vert.x event loop group here
+          // since the timer thread is got blocked between events(via sleep call)
+          return super.timer(threadFactory);
```

Comment:
I believe you could keep only the two methods above.

Reply:
Agree. Done